Neuro/Minigames/Solvers/SafeSolver.cs

```diff
@@ -0,0 +1,153 @@
+            InGameCursor.Instance.StartHoldingLMB(minigame);
+            for (float t = 0; t < TumblerSpeed; t += Time.deltaTime)
+            {
+                // exit if we find the solution
+                if (TumblerAngleNear(minigame.Tumbler.transform.eulerAngles.z + 45f, minigame.lastTumDir, (float)minigame.combo[0] * 45, 3f))
+                {
+                    InGameCursor.Instance.StopHoldingLMB();
+                    minigame.CheckTumblr(0f, minigame.Tumbler.transform.eulerAngles.z, 0, minigame.combo[0] * 45);
+                    break;
+                }
+
+                TumblerAngle = Mathf.Lerp(currTumblerAngle, newTumblerAngle, t / TumblerSpeed);
+                InGameCursor.Instance.SnapToPositionOnCircle(minigame.Tumbler, TumblerRadius, TumblerAngle);
+                yield return null;
+            }
+
+            InGameCursor.Instance.StopHoldingLMB();
+
+            // Combo 2
+            currTumblerAngle = TumblerAngle;
+            newTumblerAngle = TumblerAngle + 360f;
+            InGameCursor.Instance.StartHoldingLMB(minigame);
+            for (float t = 0; t < TumblerSpeed; t += Time.deltaTime)
+            {
+                // exit if we find the solution
+                if (TumblerAngleNear(minigame.Tumbler.transform.eulerAngles.z + 45f, minigame.lastTumDir, (float)minigame.combo[1] * 45, 3f))
+                {
+                    InGameCursor.Instance.StopHoldingLMB();
+                    minigame.CheckTumblr(0f, minigame.Tumbler.transform.eulerAngles.z, 1, minigame.combo[1] * 45);
+                    break;
+                }
+
+                TumblerAngle = Mathf.Lerp(currTumblerAngle, newTumblerAngle, t / TumblerSpeed);
+                InGameCursor.Instance.SnapToPositionOnCircle(minigame.Tumbler, TumblerRadius, TumblerAngle);
+                yield return null;
+            }
+
+            InGameCursor.Instance.StopHoldingLMB();
+
+            // Combo 3
+            currTumblerAngle = TumblerAngle;
+            newTumblerAngle = TumblerAngle - 360f;
+            InGameCursor.Instance.StartHoldingLMB(minigame);
+            for (float t = 0; t < TumblerSpeed; t += Time.deltaTime)
+            {
+                // exit if we find the solution
+                if (TumblerAngleNear(minigame.Tumbler.transform.eulerAngles.z + 45f, minigame.lastTumDir, (float)minigame.combo[2] * 45, 3f))
+                {
+                    InGameCursor.Instance.StopHoldingLMB();
+                    minigame.CheckTumblr(0f, minigame.Tumbler.transform.eulerAngles.z, 2, minigame.combo[2] * 45);
+                    break;
+                }
+
+                TumblerAngle = Mathf.Lerp(currTumblerAngle, newTumblerAngle, t / TumblerSpeed);
+                InGameCursor.Instance.SnapToPositionOnCircle(minigame.Tumbler, TumblerRadius, TumblerAngle);
+                yield return null;
+            }
+
+            InGameCursor.Instance.StopHoldingLMB();
+        }
+
+
+        // Spinner
```

Comment:
Nit: Split up the method into dedicated methods for the tumbler and the spinner; don't use comments to denote the separate sections.

Reply:
https://github.com/VedalAI/neuro-amongus/pull/76/commits/40236548ae05631f67b036ba8689bda8a6a91939